modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/CommonScriptMessageContext.java

```diff
@@ -0,0 +1,848 @@
+            axis2smc.getAxis2MessageContext().getOperationContext().setProperty(key, value);
+        }
+    }
+
+    @Override
+    public void removeProperty(String key, String scope) {
+        if (scope == null || XMLConfigConstants.SCOPE_DEFAULT.equals(scope)) {
+            Set pros = mc.getPropertyKeySet();
+            if (pros != null) {
+                pros.remove(key);
+            }
+        } else if (XMLConfigConstants.SCOPE_AXIS2.equals(scope)) {
+            //Removing property from the Axis2 Message Context
+            Axis2MessageContext axis2smc = (Axis2MessageContext) mc;
+            org.apache.axis2.context.MessageContext axis2MessageCtx = axis2smc.getAxis2MessageContext();
+            axis2MessageCtx.removeProperty(key);
+
+        } else if (XMLConfigConstants.SCOPE_TRANSPORT.equals(scope)) {
+            // Removing transport headers
+            Axis2MessageContext axis2smc = (Axis2MessageContext) mc;
+            org.apache.axis2.context.MessageContext axis2MessageCtx = axis2smc.getAxis2MessageContext();
+            Object headers = axis2MessageCtx.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+            if (headers != null && headers instanceof Map) {
+                Map headersMap = (Map) headers;
+                headersMap.remove(key);
+            }
+        } else if (XMLConfigConstants.SCOPE_OPERATION.equals(scope)) {
+            // Removing operation scope headers
+            Axis2MessageContext axis2smc = (Axis2MessageContext) mc;
+            org.apache.axis2.context.MessageContext axis2MessageCtx = axis2smc.getAxis2MessageContext();
+            OperationContext axis2oc = axis2MessageCtx.getOperationContext();
+            axis2oc.removeProperty(key);
+        }
+
+    }
+
+    private void handleSpecialProperties(String key, Object value,
+            org.apache.axis2.context.MessageContext messageContext) {
+        if (org.apache.axis2.Constants.Configuration.MESSAGE_TYPE.equals(key)) {
+            messageContext.setProperty(org.apache.axis2.Constants.Configuration.CONTENT_TYPE, value);
+            Object o = messageContext.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+            Map _headers = (Map) o;
+            if (_headers != null) {
+                _headers.remove(HTTP.CONTENT_TYPE);
+                _headers.put(HTTP.CONTENT_TYPE, value);
+            }
+        }
+    }
+
+    public Set getPropertyKeySet() {
+        return mc.getPropertyKeySet();
+    }
+
+    public Mediator getMainSequence() {
+        return mc.getMainSequence();
+    }
+
+    public Mediator getFaultSequence() {
+        return mc.getFaultSequence();
+    }
+
+    public Mediator getSequence(String key) {
+        return mc.getSequence(key);
+    }
+
+    public OMElement getFormat(String s) {
+        return null;  //To change body of implemented methods use File | Settings | File Templates.
```

Comment:
`getFormat` always return null. Is this the expected behaviour. Shall we also get rid of the auto generated comment?